pkg/operator/ceph/object/bucket/rgw-handlers.go

```diff
@@ -47,15 +48,11 @@ func (p *Provisioner) createCephUser(username string) (accKey string, secKey str
 	return u.Keys[0].AccessKey, u.Keys[0].SecretKey, nil
 }
 
-func (p *Provisioner) genUserName(obcName, obcNamespace string) (string, error) {
-	// user name can be deterministically generated by obc name and namespace. since there can't be
-	// 2 obcs in the same namespace w/ the same name, this won't collide w/in the same k8s cluster.
-	// However, it can collide in multisite setups, so use the Ceph cluster FSID
-	fsid := p.clusterInfo.FSID
-	if p.clusterInfo.FSID == "" {
-		return "", errors.Errorf("failed to find ceph cluster FSID")
-	}
-	return "obc-" + obcNamespace + "-" + obcName + "-" + fsid, nil
+func (p *Provisioner) genUserName(obc *v1alpha1.ObjectBucketClaim) string {
+	// user name can be deterministically generated from the OBC's UID. We
+	// cannot simply use the OBC's namespace and name, because they can be
+	// reused, while the preceding bucket might be retained by reclaimPolicy.
+	return "obc-" + string(obc.UID)
```

Comment:
Personally, I like the idea of keeping `obcNamespace` and `obcName` here to help with visual inspection of users on the backend and mapping them into the k8s domain. This isn't a hard requirement, but I believe it's a worthwhile "hint" that can help when doing complex debugging, and it's essentially free.

```suggestion
	return "obc-" + obc.Namespace + "-" obc.Name + string(obc.UID)
```